### DISCUSSION TITLE
Hydration question: anchor cannot have flow children?

### DISCUSSION THREAD

**Discussion by nlfiedler** [leptos-rs/leptos#3359] [General]
This fails with a hydration error on the top `div` element:

```html
        <div class="grid is-col-min-16 padding-2">
            <For
                each=move || results.get_value().into_iter().enumerate().map(|(i, r)| StoredValue::new((i, r)))
                key=|r| r.get_value().1.asset_id
                let:elem
            >
                <div class="cell">
                    <a on:click=move |_| onclick(elem.get_value().0)>
                        <div class="card">
                            <div class="card-image">
                                <figure class="image">
```

If I remove the anchor tag then the error goes away. But can't anchor tags have flow content in HTML 5? Is this a bug in the hydration, or am I doing something wrong?

**Comment by gbj** [comment]
To help clarify the error message: the error lists the first element where the framework *notices* that there's a hydration error, i.e., when it's expecting an element and gets a text node, or some other mismatch. If it's listing the first `<div>`, you may need to go up and back (to that div's parents and the siblings that come before it) to look for the issue)

I don't see anything obviously wrong when I test out this snippet as a regular HTML file, no.

**Comment by nlfiedler** [comment]
Removing the anchor tag makes the error go away, would that imply that it is the cause of the problem?

**Comment by nlfiedler** [comment]
A confounding aspect of the hydration errors is that they do not occur every time, so as I try adding and removing different elements, sometimes the error occurs and sometimes it doesn't, so I'm having a very difficult time debugging this.

**Reply by gbj** [reply]
If the hydration error has to do with the HTML structure itself I am pretty sure it will happen every time.
